fem/nonlinearform.hpp

```diff
@@ -363,8 +363,39 @@ class BlockNonlinearForm : public Operator
                              Array<int> &bdr_marker)
    { bfnfi.Append(nlfi); bfnfi_marker.Append(&bdr_marker); }
 
-   virtual void SetEssentialBC(const Array<Array<int> *>&bdr_attr_is_ess,
-                               Array<Vector *> &rhs);
+   /** @brief Set essential boundary conditions to each finite element space
+       using boundary attribute markers.
+    
+       This method calls `FiniteElementSpace::GetEssentialTrueDofs()` for each
+       space and stores ess_tdof_lists internally.
+    
+       If `rhs` vectors are not `NULL`, the entries corresponding to these
+       essential DoFs are set to zero. This ensures compatibility with the
+       output of the `Mult()` method, which also zeroes out these entries.
+    
+       @param[in] bdr_attr_is_ess A list of boundary attribute markers for each
+       space.
+       @param[in,out] rhs         An array of optional right-hand side vectors.
+       If a vector at `rhs[i]` is not `nullptr`, its essential DoFs will be set
```

Comment:
There is mixed use of `NULL` and `nullptr` in these comments, and either could be used. Maybe just the vague English term non-null would be more general and less confusing?